app/models/site_configuration.rb

```diff
@@ -3,14 +3,66 @@ class SiteConfiguration < ApplicationRecord
   before_create :ensure_single_record
   validate :validate_help_link_items
 
+  has_many :revision_reasons
+
+  accepts_nested_attributes_for :revision_reasons, allow_destroy: true
+  validate :max_undiscarded_revision_reasons
+  validate :max_revision_reasons
+
   HELP_LINK_KEYS = %w[get_started_link_item best_practices_link_item dictionary_link_item user_guide_link_item]
 
   def self.instance
     first_or_create
   end
 
+  def revision_reasons_attributes=(attributes)
+    attributes.each do |attribute, _|
+      if attribute["_destroy"] == true
```

Comment:
The `_destroy`, method is used by rails active record to mark nested attributes for destruction. If the goal is to archive, use a different property name to keep with convention. Better yet, have a`should_discard` or similar attribute accessor in the `revision_reason` model, and pass this flag from the front-end. In the `revision_reason` model have a `before_save` callback to set the `discarded_at` if this attribute is `true`

Reply:
Isnt the idea to override the existing destruction behaviour? If I go RevisionReason.first.destroy, that would typically destroy the record, but the replaced behaviour is already abstracted away for me. Why would I need to make this distinction? Wouldnt this create ambiguisty?